pennylane/templates/subroutines/qsvt.py

```diff
@@ -371,7 +371,7 @@ def compute_matrix(*args, **kwargs):
                     op_list.append(adjoint(UA_copy))
 
             op_list.append(projectors[-1])
-            mat = qml.matrix(qml.prod(*tuple(op_list)))
+            mat = qml.matrix(qml.prod(*tuple(op_list[::-1])))
```

Comment:
Is this the solution to the issue @Jaybsoni was mentioning yesterday?

Reply:
:thinking: I'm not sure. We did discuss the solution in this PR and I believe @Jaybsoni and I are on the same page.

Reply:
Yes, previously, without the `[::-1]`, the matrix computed in the `qml.prod` was of the opposite order, which was causing the matrix computation mismatch.

Reply:
Yes @trbromley, 

There was a typo in the test/example I showed with the order of the operations. This fixes that and ensures everything produces the same results 👍🏼